pallets/multisig/src/lib.rs

```diff
@@ -117,6 +117,54 @@ pub trait Trait: frame_system::Trait + IdentityTrait {
     type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
 }
 
+/// Details of a multisig proposal
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct ProposalDetails<T> {
+    /// Number of yes votes
+    pub approvals: u64,
+    /// Number of no votes
+    pub rejections: u64,
+    /// Status of the proposal
+    pub status: ProposalStatus,
+    /// Expiry of the proposal
+    pub expiry: Option<T>,
+    /// Should the proposal be closed after getting inverse of sign required reject votes
+    pub auto_close: bool,
+}
+
+impl<T: core::default::Default> ProposalDetails<T> {
+    /// Create new `Asset` with the given reference to the client.
+    pub fn new(expiry: Option<T>, auto_close: bool) -> Self {
+        Self {
+            status: ProposalStatus::ActiveOrExpired,
+            expiry,
+            auto_close,
+            ..Default::default()
+        }
+    }
+}
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq)]
+/// Status of a multisig proposal
+pub enum ProposalStatus {
+    /// Proposal does not exist
+    Invalid,
+    /// Proposal has not been closed yet. This means that it's either expired or open for voting.
+    ActiveOrExpired,
```

Comment:
IMHO having two separate statuses, one for Active and another for Expired, would remove the need to do additional case analysis (e.g., if expired then give an error, otherwise proceed).

Reply:
Problem is that we can not update the status from active to expired when the proposal expires. To do that, we'd have to do something like we do in the bridge module and update statuses at the start of every block. I don't think it's worth adding this extra load on the chain.